cmd/mposter/main.go

```diff
@@ -0,0 +1,139 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"text/template"
+)
+
+func main() {
+	params, err := parseParams(os.Args)
+	if nil != err {
+		log.Fatal(err)
+	}
+
+	err = run(params)
+	if nil != err {
+		log.Fatal(err)
+	}
+}
+
+func parseParams(params []string) (runParams, error) {
+
+	//TODO: support reading to and writing from a file
+	result := newRunParams()
+
+	commandLine := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	commandLine.StringVar(&result.url, "url", "http://localhost:8080/example/", "url to post to") //TODO: no default and fail if not specified.
+	commandLine.StringVar(&result.fieldSeparator, "separator", "", "row field separator. White space if not specified.")
+	commandLine.IntVar(&result.stopOnErrorCount, "stop-on-err-count", result.stopOnErrorCount, "Stop on consequent error results")
+
+	return result, commandLine.Parse(os.Args[1:])
+}
+
+func run(params runParams) error {
+
+	scanner := bufio.NewScanner(params.input)
+
+	templated := strings.Contains(params.url, "{{")
+
+	var paramsToUrl func(string) (string, error)
+	if templated {
+		urlTemplate, err := template.New("url").Parse(params.url)
+		if err != nil {
+			return fmt.Errorf("parse url template \"%s\": %w", params.url, err)
+		}
+
+		paramsToUrl = func(line string) (string, error) {
+			row := splitRows(line, params.fieldSeparator)
+			//TODO: explicit param or guessing for whether it's a path or not.
```

Comment:
TODO: guess or explicitly specify the parameters of the template url or path encoded should be.